packages/modules/create-imodel/README.md

```diff
@@ -4,6 +4,28 @@
 
 Contains components to create or update an iModel.
 
+### iTwinUI V3 breaking change
```

Comment:
This does not look good.
We have N packages using iTwinUI and wanting to migrate to v3 independently.
v3 was advertised to work easily with v2 without any huge changes. Installing alias is a huge change, IMO.
@mayank99 @FlyersPh9, before the big wave of migration started, could you have a look if its possible to make this easier? Cause adding documentation on each package to add some external css to work, does not look good.

Reply:
it's complicated, we have to make a tradeoff between:
- relying on side-effect imports (which causes issues with ordering, duplication and a lot more)
- relying on client-side JS to inject styles (which is hacky, doesn't work with SSR, and gives no control to the end user for postprocessing)
- providing a full stylesheet (which requires manual work)

the ideal scenario here is that applications will migrate to v3 earlier, because they don't really have to wait for packages anymore

Reply:
Application cannot migrate to v3 earlier than package. Scope is just bigger compared to the package.

Reply:
then the application could continue using the old version of all packages that are already compatible with iTwinUI v2.

it's similar to how an application would not consume a dependency that uses react 18 features, until the application itself is updated to use react 18.

in v2, packages had to migrate first, which was a blocker for applications. in v3, the relationship is flipped: applications can migrate to v3 first.

---

i understand the scope is bigger, but it needs to be done at some point anyway. v2 will not get any new features.

no matter what we do, application developers will always come up with excuses and complaints. we undid all breaking changes in itwinui-react v2, and yet so many of our users are still using v1. it feels like a one-way exchange.

i also don't see npm aliases as a "huge change". it's two lines of code. what am i missing? maybe the biggest concern here is unfamiliarity. we could try documenting this in more detail if that helps. (see [current documentation](https://github.com/iTwin/iTwinUI/wiki/iTwinUI-react-v3-migration-guide#styles))

Reply:
Well, if as you mentionned using different aliases causes context to unmatch, this is a huge change because it will likely give weird problems. (Still need to be confirmed though)

Reply:
it won't cause any problems if packages keep using ThemeProvider in their entrypoints. the application will only need to import styles (one line in `package.json` and one line in `App.tsx`).

Reply:
@veekeys @raplemie would you be able to help me test this PR within one or more applications? https://github.com/iTwin/iTwinUI/pull/1734

i mainly want to see if bundlers choke on the dynamic `import('../../../styles.css')`.

Reply:
Mind publishing beta? in v2 preferably?

the fallback from cdn would probably hit out CSP too.. But hoping it would never reach that, so its fine (document it after in case will go with this one).

Reply:
it will not be possible to publish in v2, because the CSS strategy is completely different there.

for the purposes of admin-components though, there is no change needed yet, so this PR is good to merge.

after admin-components is released, you could test out with itwinui v3

Reply:
You want me to make a local build and link it then? Or what would be the best here.. Maybe publish beta v3 then?

Reply:
my initial thought was to test with these two things:
- new admin-components release that includes #112 and #113
- new itwinui-react prerelease that includes https://github.com/iTwin/iTwinUI/pull/1734

however, if you are reverting to v2 (#114), then you'll have nowhere to test v3.

Reply:
Tested in storybook on this branch (removed the styles import in preview.js) (webpack 5, with storybook magic)
Also tested in appui connected tests apps updating imodel-browser-react to 1.2.0 and forcing `imodel-browser-react`->`itwinui-react` dependency to `3.1.3-dev.1` through `.pnpmfile.cjs` (webpack 5, with CRA magic (bentley scripts))

Styles are applied correctly, was coming from 0.13.3, which only required `projectId`->`iTwinId` breaking change introduced by `1.0.0`. (That was expected)

Tested in web and electron (which uses the same webpacked frontend...) only thing I could see for now is this, which is apparently triggered by a `Tile` (see other image) but all the tiles are wrapped in a `ThemeProvider`, so I'm not sure what's going on. I'll look a bit more tomorrow, I'm a bit surprised that the classname is `iui-tile` and not `_iui3-tile`...

<img width="615" alt="image" src="https://github.com/iTwin/admin-components-react/assets/1904889/3e20dc78-ec12-4ca3-88fd-d4e0000986d6">

![image](https://github.com/iTwin/admin-components-react/assets/1904889/c8302930-e633-416d-959e-4122c739ebaf)

Reply:
the `"iui-tile"` thing is expected — we transform the CSS class at runtime, so don't worry about that.

it's possible that there are two different modules being used (ESM vs CJS maybe?), but if you're having trouble fixing it and the theming works, then this error can be ignored because it only happens during development.

Reply:
Ok, after more research on this, the breakpoint was not on the right spot in the above image, the issues come from `NoResults` component which is indeed missing the `ThemeProvider`. Adding it in this PR shoudl fix that last issue.

Reply:
This is now released for stable use in [3.2.0](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.2.0)

Reply:
Thanks for this, I'll look at reincorporating all the reverted changes !